### DISCUSSION TITLE
Combine Stores

### DISCUSSION THREAD

**Discussion by zakharchanka** [Kotlin/kmm-production-sample#32] [General]
Hello. Thank you for nice sample! But it is not clear for me how to combine stores using your architecture?
For example there are 3 different stores that depends on one common request.

**Comment by terrakok** [comment]
I guess this stores should be childs of common store which handles common requests

**Reply by zakharchanka** [reply]
But then the principle of unidirectional data flow is violated. Don't you think so?

**Reply by terrakok** [reply]
Why?
Unidirectional data flow exists between logic -> state -> UI
https://code-cartoons.com/a-cartoon-intro-to-redux-3afb775501a6

we can split reducers for separating complex logic

**Reply by zakharchanka** [reply]
Yes, I agree with you. We can split reducers, because reducers don't have any state. But stores have, so putting store inside another store seems like a bad idea for me. Correct me if I'm wrong.
